sympy/polys/polytools.py

```diff
@@ -5307,6 +5315,182 @@ def half_gcdex(f, g, *gens, **args):
         return s, h
 
 
+def _gcdex_steps_domain(a, b, K):
+    """
+    Generator for intermediate steps in the extended euclidean algorithm
+    on domain elements. Helper function for `gcdex_steps`.
+    """
+    if not K.is_PID:
+        raise DomainError("gcdex_steps is only for Euclidean domains")
+
+    s1, s2 = K.one, K.zero
+    t1, t2 = K.zero, K.one
+
+    while b:
+        yield s2, t2, b
+
+        quotient, remainder = K.div(a, b)
+
+        a, b = b, remainder
+        s1, s2 = s2, s1 - quotient * s2
+        t1, t2 = t2, t1 - quotient * t2
+
+
+def _gcdex_steps_polynomial(f, g, auto):
+    """
+    Generator for intermediate steps of the extended euclidean algorithm
+    on polynomials. Helper function for `gcdex_steps`.
+    """
+    if auto and f.get_domain().is_Ring:
+        f, g = f.to_field(), g.to_field()
+
+    if not f.get_domain().is_PID:
+        raise DomainError("gcdex_steps is only for Euclidean domains")
+
+    if not f.is_univariate:
+        raise ValueError('univariate polynomial expected')
```

Comment:
I think that the `is_PID` check here should really be `if f.domain.is_Field`. Univariate polynomials over a field are suitable for `gcdex`.